src/os/windows/rand.cpp

```diff
@@ -9,40 +9,39 @@
 /**
  * \file Windows-specific implementation of rand.h
  */
-#include <time.h>
-#include <stdlib.h>
-#include <process.h>
 
+#include <Winsock2.h>
+#include <windows.h>
+#include <ctime>
+#include <cstdlib>
+#include <process.h>
+#include <bcrypt.h>
 #include "../rand.h"
 #include "../lib.h"
 
-// See the MSDN documentation for RtlGenRandom. We will try to load it
-// and if that fails, use our own random number generator. The function
-// call is indirected through a function pointer
-
-static BOOLEAN (APIENTRY *pfnGetRandomData)(void*, ULONG) = NULL;
 
 bool pws_os::InitRandomDataFunction()
 {
-  HMODULE hLib = HMODULE(pws_os::LoadLibrary(_T("ADVAPI32.DLL"), loadLibraryTypes::SYS));
-
-  BOOLEAN (APIENTRY *pfnGetRandomDataT)(void*, ULONG) = NULL;
-  if (hLib != NULL) {
-    pfnGetRandomDataT =
-      (BOOLEAN (APIENTRY *)(void*,ULONG))pws_os::GetFunction(hLib, "SystemFunction036");
-    if (pfnGetRandomDataT) {
-      pfnGetRandomData = pfnGetRandomDataT;
-    }
+  // Check if BCryptGenRandom is available
+  HMODULE hBcrypt = (HMODULE)pws_os::LoadLibrary(TEXT("bcrypt.dll"), loadLibraryTypes::SYS);
+  if (hBcrypt == nullptr) {
+    return false; // BCryptGenRandom not available
   }
-  return (hLib != NULL && pfnGetRandomDataT != NULL);
+  // Check if BCryptGenRandom function is available
+  FARPROC pFunc = GetProcAddress(hBcrypt, "BCryptGenRandom");
+  if (pFunc == nullptr) {
```

Comment:
Suggestion: remove this if block and replace line 34 with
```
return pFunc != nullptr;
```
so the `FreeLibrary` invocation is in a single place.

Reply:
Thanks!